### PR TITLE
Provide `keep_output_port` attribute to `add_opoutput` function

### PR DESCRIPTION
**Pull request by achetver**
Root cause analysis: Middle transformation `ConvertGroupedStridedSlice` creates `Result` operations connected to `VariadicSplit` operations. These `Result` operations not erased at IR emitting stage as we expect.

Solution: Add providing of `keep_output_port` attribute to `add_opoutput` function. This attribute determines will `Result` operation be saved in IR or not. 

Ticket: 57628

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests: Updated;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: Done.

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

**Review by vgavrilo — APPROVED** [review]

**Review by yekruglov — APPROVED** [review]

**Comment by achetver** [comment]
@rkazants Please, review this PR.

File: model-optimizer/mo/graph/graph.py [hunk 1]
```diff
@@ -1032,21 +1032,24 @@ def dict_includes(big: dict, sub_dict: dict, skip_attr_names=[]):
     )
 
 
-def add_opoutput(graph: Graph, node_name: str, port: int, cut: bool = True):
+def add_opoutput(graph: Graph, node_name: str, port: int, cut: bool = True, keep_output_port=None):
```
Comment (rkazants) [inline model-optimizer/mo/graph/graph.py:1035]: keep_output_port=None -> keep_output_port: bool = False

File: model-optimizer/mo/graph/graph.py [hunk 2]
```diff
@@ -1032,21 +1032,24 @@ def dict_includes(big: dict, sub_dict: dict, skip_attr_names=[]):
...
     """
     Creates and connects Result node to node_name port. Cuts existing port if requested.
     :param graph: graph to operate with
     :param node_name: name of existing node in the graph that we want to add Result to
     :param port: output port of node to connect Result to
     :param cut: determines way of operating with edge specified by node_name and port
+    :param keep_output_port: special attribute determines will this operation be saved in IR or not
```
Comment (rkazants) [inline model-optimizer/mo/graph/graph.py:1042]: :param keep_output_port: special attribute determines if this operation be saved in IR or not

**Comment by rkazants** [comment]
@lazarevevgeny, I propose to run DLbenchmark validation if it hasn't been yet.

File: model-optimizer/unit_tests/extensions/middle/ConvertGroupedStridedSlice_test.py [hunk 3]
```diff
@@ -60,8 +60,8 @@
     'concat_1_data': {'value': None, 'shape': None, 'kind': 'data'},
 
     'op_output': {'kind': 'op', 'op': 'Result'},
-    'op_output_1': {'kind': 'op', 'op': 'Result'},
-    'op_output_2': {'kind': 'op', 'op': 'Result'},
+    'op_output_1': {'kind': 'op', 'op': 'Result', 'keep_output_port': True},
+    'op_output_2': {'kind': 'op', 'op': 'Result', 'keep_output_port': True},
```
Comment (rkazants) [inline model-optimizer/unit_tests/extensions/middle/ConvertGroupedStridedSlice_test.py:65]: also, add case with 'keep_output_port' equal to `False`

File: model-optimizer/mo/graph/graph.py (hunk 1 above)
Comment (achetver) [reply model-optimizer/mo/graph/graph.py:1035]: I used `None` value to not change logic - this attribute either exist and has value True or does not exist at all. So I tried to minimize impact for existing code.

PR with this attribute implementation: https://github.com/openvinotoolkit/openvino/pull/5136

**Comment by achetver** [comment]
@achetver We check impact to validation using IR's from nightly tests and DL benchmark. You can find the results in the ticket. And we have approve from Accuracy Checker team to merge this fix to master branch.

File: model-optimizer/unit_tests/extensions/middle/ConvertGroupedStridedSlice_test.py (hunk 3 above)
Comment (achetver) [reply model-optimizer/unit_tests/extensions/middle/ConvertGroupedStridedSlice_test.py:65]: See my comment above.

File: model-optimizer/mo/graph/graph.py (hunk 2 above)
Comment (achetver) [reply model-optimizer/mo/graph/graph.py]: Done.

File: model-optimizer/mo/graph/graph.py (hunk 1 above)
Comment (rkazants) [reply model-optimizer/mo/graph/graph.py:1035]: After your change 'keep_output_port' attribute with `None` value always exists in `Result` node. And the logic for handling such case with `None` value probably exists. In PR#5136 I observe `False` value. Please check if these two cases (`False` and `None`) can be emerged. I dislike that we may have three cases (true, false, and none) that makes solution not transparent. Please align the solution.

Comment (lazarevevgeny) [reply model-optimizer/mo/graph/graph.py:1035]: I vote for the default value `False` instead of `None`

Comment (achetver) [reply model-optimizer/mo/graph/graph.py]: Ok, default value changed.

File: model-optimizer/unit_tests/extensions/middle/ConvertGroupedStridedSlice_test.py (hunk 3 above)
Comment (achetver) [reply model-optimizer/unit_tests/extensions/middle/ConvertGroupedStridedSlice_test.py:65]: Case with `keep_output_port` value equal to `False` is not applicable to this function. And test for `add_opoutput` with default value `keep_output_port` = `False` already exist.

File: model-optimizer/mo/graph/graph.py [hunk 4]
```diff
@@ -1032,21 +1032,24 @@ def dict_includes(big: dict, sub_dict: dict, skip_attr_names=[]):
     )
 
 
-def add_opoutput(graph: Graph, node_name: str, port: int, cut: bool = True):
+def add_opoutput(graph: Graph, node_name: str, port: int, cut: bool = True, keep_output_port=False):
```
Comment (rkazants) [inline model-optimizer/mo/graph/graph.py:1035]: keep_output_port: bool = False

Comment (achetver) [reply model-optimizer/mo/graph/graph.py]: Added.

**Review by rkazants — APPROVED** [review]
please make CI green

**Comment by achetver** [comment]
> please make CI green

Now CI is green.

**Pull request merged**